src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -222,6 +268,29 @@ export class SavedObjectsClient {
     };
   }
 
+  _withKibanaIndexAndMappingFallback(method, params, fallbackParams) {
+    const fallbacks = {
+      'create': ['is_single_type'],
+      'index': ['is_single_type'],
+      'update': ['document_missing_exception']
+    };
+
+    return new Promise((resolve, reject) => {
```

Comment:
There's no need to use the Promise constructor here.  Since `_withKibanaIndex` returns a promise itself, you can just chain off of that and return the result.